[Bug] r.texture: test_sv fails on Mac with arm64

## Describe the bug
<!-- A clear and concise description of what the bug is. -->
One test on the `r.texture` module failed when running the sv method on MacOS in PR #3786. The values are different from those from WSL. Actually, the other methods derive slightly different values of no more than 1e-2. This error can be reproduced on my own Macbook with an M1 chip. 

See the detailed testing result below:

<details>

```
========================================================================
...........F.
======================================================================
FAIL: test_sv (__main__.TestRasterreport.test_sv)
Testing method sv
----------------------------------------------------------------------
Traceback (most recent call last):
  File "raster/r.texture/testsuite/test_texture.py", line 133, in test_sv
    self.assertRasterFitsUnivar(output, reference=values, precision=1e-2)
  File "etc/python/grass/gunittest/case.py", line 289, in assertRasterFitsUnivar
    self.assertModuleKeyValue(
  File "etc/python/grass/gunittest/case.py", line 271, in assertModuleKeyValue
    self.fail(self._formatMessage(msg, stdMsg))
AssertionError: r.univar map=sv_SV percentile=90.0 nprocs=1 separator== -g difference:
mismatch values (key, reference, actual): [('max', 45368492, 45368496), ('mean', 2248724.35829364, 2248724.38215788), ('variance', 2332049431762.5, 2332049495199.41)]
command: r.univar map=sv_SV percentile=90.0 nprocs=1 separator== -g {'map': 'sv_SV', 'separator': '=', 'flags': 'g'}

----------------------------------------------------------------------
Ran 13 tests in 17.564s
FAILED (failures=1)
========================================================================
FAILED ./raster/r.texture/testsuite/test_texture.py (1 test failed)
```


</details>


## To reproduce
<!-- Describe the steps to reproduce the behavior, e.g.:
-->

1. Fetch the `test_texture.py` file in PR #3786. 
2. Use `precision=1e-2` for every unit test function.
3. Run this test file on an arm-based macOS machine.


## Expected behavior
<!-- If not obvious, describe what you expected to happen. -->
Only test_sv fails, and others pass. It should show:
```
mismatch values (key, reference, actual): [('max', 45368492, 45368496), ('mean', 2248724.35829364, 2248724.38215788), ('variance', 2332049431762.5, 2332049495199.41)]
```
 If you use a smaller precision, you may see more tests fail.

## System description
<!--
- Operating System: [e.g. Windows, Linux, ..., incl. version]
- GRASS GIS version: [e.g. 4.0]

- details about further software components
    - run `g.version -rge` in a GRASS GIS terminal session or check in
      the GUI menu "Help > About"
    - run `python3 -c "import sys, wx; print(sys.version); print(wx.version())"`
      to print the Python and wxPython version numbers
-->
OS: macOS (arm chip)
GRASS GIS version: 8.4.0

## Additional context
<!-- Add any other context about the problem. -->
#3395, #3397, and #3398 reported similar/same bug. Not sure if this bug happens in `r.texture` or `r.univar`.

--- a/.github/workflows/macos_install.sh
+++ b/.github/workflows/macos_install.sh
@@ -59,8 +59,8 @@ CONFIGURE_FLAGS="\
   --without-x \
 "
 
-export CFLAGS="-O2 -pipe -ffp-contract=fast -arch ${CONDA_ARCH} -DGL_SILENCE_DEPRECATION -Wall -Wextra -Wpedantic -Wvla"
-export CXXFLAGS="-O2 -pipe -ffp-contract=fast -stdlib=libc++ -arch ${CONDA_ARCH} -Wall -Wextra -Wpedantic"
+export CFLAGS="-O2 -pipe -ffp-contract=off -arch ${CONDA_ARCH} -DGL_SILENCE_DEPRECATION -Wall -Wextra -Wpedantic -Wvla"
+export CXXFLAGS="-O2 -pipe -ffp-contract=off -stdlib=libc++ -arch ${CONDA_ARCH} -Wall -Wextra -Wpedantic"
 export CPPFLAGS="-isystem${CONDA_PREFIX}/include"
 
 ./configure $CONFIGURE_FLAGS

--- a/.github/workflows/macos_install.sh
+++ b/.github/workflows/macos_install.sh
@@ -59,8 +59,8 @@ CONFIGURE_FLAGS="\
   --without-x \
 "
 
-export CFLAGS="-O2 -pipe -arch ${CONDA_ARCH} -DGL_SILENCE_DEPRECATION -Wall -Wextra -Wpedantic -Wvla"
-export CXXFLAGS="-O2 -pipe -stdlib=libc++ -arch ${CONDA_ARCH} -Wall -Wextra -Wpedantic"
+export CFLAGS="-O2 -pipe -ffp-contract=fast -arch ${CONDA_ARCH} -DGL_SILENCE_DEPRECATION -Wall -Wextra -Wpedantic -Wvla"
+export CXXFLAGS="-O2 -pipe -ffp-contract=fast -stdlib=libc++ -arch ${CONDA_ARCH} -Wall -Wextra -Wpedantic"
 export CPPFLAGS="-isystem${CONDA_PREFIX}/include"
 
 ./configure $CONFIGURE_FLAGS

--- a/.github/workflows/ubuntu.yml
+++ b/.github/workflows/ubuntu.yml
@@ -130,9 +130,9 @@ jobs:
       - name: Build
         env:
           # TODO: -pedantic-errors here won't go through ./configure (with GNU C)
-          CFLAGS: -fPIC -Wvla
+          CFLAGS: -fPIC -Wvla -ffp-contract=off
           # TODO: -pedantic-errors here won't compile
-          CXXFLAGS: -fPIC
+          CXXFLAGS: -fPIC -ffp-contract=off
         run: .github/workflows/build_${{ matrix.config }}.sh $HOME/install -Werror
 
       - name: Add the bin directory to PATH

--- a/raster/r.texture/testsuite/test_rtexture.py
+++ b/raster/r.texture/testsuite/test_rtexture.py
@@ -127,21 +127,13 @@ class TestRasterreport(TestCase):
         basename = "SV"
         method = "sv"
         output = f"{method}_{basename}"
-        # The results on macOS is slightly different from the other platforms
-        if IS_MAC:
-            values = """min=0
-            max=45368496
-            mean=2248724.38215788
-            variance=2332049495199.41
-            n=996244"""
-        else:
-            values = """min=0
-            max=45368492
-            mean=2248724.35829364
-            variance=2332049431762.5
-            n=996244"""
+        values = """min=0
+        max=45368492
+        mean=2248724.35829364
+        variance=2332049431762.5
+        n=996244"""
         self.assertModule("r.texture", input=self.input, output=method, method=method)
-        self.assertRasterFitsUnivar(output, reference=values, precision=1e-2)
+        self.assertRasterFitsUnivar(output, reference=values, precision=1e-4)
 
     def test_se(self):
         """Testing method se"""

--- a/raster/r.texture/testsuite/test_rtexture_parallel.py
+++ b/raster/r.texture/testsuite/test_rtexture_parallel.py
@@ -140,23 +140,15 @@ class TestRasterreport(TestCase):
         basename = "SV"
         method = "sv"
         output = f"{method}_{basename}"
-        # The results on macOS is slightly different from the other platforms
-        if IS_MAC:
-            values = """min=0
-            max=45368496
-            mean=2248724.38215788
-            variance=2332049495199.41
-            n=996244"""
-        else:
-            values = """min=0
-            max=45368492
-            mean=2248724.35829364
-            variance=2332049431762.5
-            n=996244"""
+        values = """min=0
+        max=45368492
+        mean=2248724.35829364
+        variance=2332049431762.5
+        n=996244"""
         self.assertModule(
             "r.texture", input=self.input, output=method, method=method, nprocs=THREADS
         )
-        self.assertRasterFitsUnivar(output, reference=values, precision=1e-2)
+        self.assertRasterFitsUnivar(output, reference=values, precision=1e-4)
 
     def test_se(self):
         """Testing method se"""

